code/WorkInProgress/TempEngine.dm

```diff
@@ -458,38 +431,46 @@
 			last = target
 			target = pick(next)
 
-
-	attack_ai(mob/user)
-		if(status & (BROKEN|NOPOWER)) return
-
-		interacted(user)
-
-	attack_hand(mob/user)
-
-		add_fingerprint(user)
-
-		if(status & (BROKEN|NOPOWER)) return
-
-		interacted(user)
-
-	proc/interacted(mob/user)
-		window.Subscribe(user.client)
-		return 1
-
-	Topic(href, href_list)
-		..()
-
-		if( href_list["close"] )
-			usr.Browse(null, "window=teg")
-			src.remove_dialog(usr)
-			return 0
-
-		return 1
-
 	power_change()
 		..()
 		updateicon()
 
+/obj/machinery/power/generatorTemp/ui_interact(mob/user, datum/tgui/ui)
+	ui = tgui_process.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "TEG", src.name)
+		ui.open()
+
+/obj/machinery/power/generatorTemp/ui_data(mob/user)
+	var/list/data = list()
+	data["output"] = src.lastgen
+	data["history"] = src.history
+	if(src.circ1)
+		data["hotCircStatus"] = src.circ1
+		data["hotInletTemp"] = src.circ1.air1.temperature
+		data["hotOutletTemp"] = src.circ1.air2.temperature
+		data["hotInletPres"] = MIXTURE_PRESSURE(src.circ1.air1) KILO PASCALS
+		data["hotOutletPres"] = MIXTURE_PRESSURE(src.circ1.air2) KILO PASCALS
+	else
+		data["hotCircStatus"] = null
+		data["hotInletTemp"] = 0
+		data["hotOutletTemp"] = 0
+		data["hotInletPres"] = 0
+		data["hotOutletPres"] = 0
```

Comment:
Could also consider doing as a nested list, then just having `null` if `src.circ1` isn't set. Would mean handling a little more in the UI, but I don't think that's necessarily a bad thing.

This is absolutely simple and readable enough that I think the `if`/`else` is absolutely fine here. Educating for more complex cases.